### PR TITLE
Old virology healing symptoms have been brought back!

### PR DESCRIPTION
**Pull request by Jujumatic**
Old virology healing symptoms have been brought back! Virology healing can become good again! 
Been wanting to do this for a while now, and it's finally here. This should make virology less useless for those people who want to be awesome and spread good stuff around the station. Update: I also brought back viral aggressive metabolism

:cl:
add: Old virology healing symptoms have been brought back! On top of this, *some* of the required reagents to get them have been adjusted, as some TG symptoms will still be present
add: Also brought back the genetic saboteur and genetic healing symptom. On top of this, viral aggressive metabolism has been revived!
tweak: Made some of the TG symptoms require different reagents to obtain, to make some of the actually useful ones easier to get, and made some symptoms unobtainable as they will be replaced by the superior old healing symptoms.
/:cl:

### CODE REVIEW

**Comment by ghost** [comment]
oh my god! amazing

**Comment by Jujumatic** [comment]
I just need to fetch the descriptions from the hippie wiki since I didn't manage to find the descriptions in oldcode for some reason. Not sure why I couldn't find it...

**Comment by Jujumatic** [comment]
>tfw erik still hasnt brought back the forums and wiki so i cant continue with this pr 

image
image
image
image

**Comment by JohnGinnane** [comment]
you can still use tg wiki which is *mostly* the same

**Comment by Jujumatic** [comment]
nope, all the old virology healing symptoms got removed and so did their descriptions as well as their bonus thresholds, effects and descriptions. and from what i checked there isn't an archived version of it either, and i couldn't find any trace of the old virology descriptions in the archived code either ???

**Comment by Carbonhell** [comment]
sure +1

**Comment by praisenarsie** [comment]
they were removed with good reason

you are honestly a chumboid if you support this pr because it's not that hard to get an op healing virus and that's a problem

**Comment by Jujumatic** [comment]
they were removed cause tg is ghey

File: hippiestation/code/datums/diseases/advance/symptoms/heal.dm [hunk 1]
```diff
@@ -0,0 +1,632 @@
...
+	return power
+
+/datum/symptom/heal/proc/Heal(mob/living/M, datum/disease/advance/A, actual_power)
+	return TRUE
+
+/datum/symptom/heal/proc/passive_message_condition(mob/living/M)
+	return TRUE
+
+
+/datum/symptom/heal/starlight
```
Comment (GuyonBroadway) [inline hippiestation/code/datums/diseases/advance/symptoms/heal.dm]: This symptom is already in game, there's no need to copy it.

File: hippiestation/code/datums/diseases/advance/symptoms/heal.dm [hunk 2]
```diff
@@ -0,0 +1,632 @@
...
+		M.reagents.remove_reagent(R.id, actual_power)
+		if(food_conversion)
+			M.nutrition += 0.3
+		if(prob(2))
+			to_chat(M, "<span class='notice'>You feel a mild warmth as your blood purifies itself.</span>")
+	return 1
+
+
+
+/datum/symptom/heal/metabolism
```
Comment (GuyonBroadway) [inline hippiestation/code/datums/diseases/advance/symptoms/heal.dm]: This symptom also is in game.

File: hippiestation/code/datums/diseases/advance/symptoms/heal.dm [hunk 3]
```diff
@@ -0,0 +1,632 @@
...
+	if(triple_metabolism)
+		C.reagents.metabolize(C, can_overdose=TRUE)
+	C.overeatduration = max(C.overeatduration - 2, 0)
+	var/lost_nutrition = 9 - (reduced_hunger * 5)
+	C.nutrition = max(C.nutrition - (lost_nutrition * HUNGER_FACTOR), 0) //Hunger depletes at 10x the normal speed
+	if(prob(2))
+		to_chat(C, "<span class='notice'>You feel an odd gurgle in your stomach, as if it was working much faster than normal.</span>")
+	return 1
+
+/datum/symptom/heal/darkness
```
Comment (GuyonBroadway) [inline hippiestation/code/datums/diseases/advance/symptoms/heal.dm]: This one exists too!.

File: hippiestation/code/datums/diseases/advance/symptoms/heal.dm [hunk 4]
```diff
@@ -0,0 +1,632 @@
...
+		if(L.heal_damage(heal_amt/parts.len, heal_amt/parts.len * 0.5)) //more effective on brute
+			M.update_damage_overlays()
+	return 1
+
+/datum/symptom/heal/darkness/passive_message_condition(mob/living/M)
+	if(M.getBruteLoss() || M.getFireLoss())
+		return TRUE
+	return FALSE
+
+/datum/symptom/heal/coma
```
Comment (GuyonBroadway) [inline hippiestation/code/datums/diseases/advance/symptoms/heal.dm]: And this one.

File: hippiestation/code/datums/diseases/advance/symptoms/heal.dm [hunk 5]
```diff
@@ -0,0 +1,632 @@
...
+		uncoma(M)
+
+	return 1
+
+/datum/symptom/heal/coma/passive_message_condition(mob/living/M)
+	if((M.getBruteLoss() + M.getFireLoss()) > 30)
+		return TRUE
+	return FALSE
+
+/datum/symptom/heal/water
```
Comment (GuyonBroadway) [inline hippiestation/code/datums/diseases/advance/symptoms/heal.dm]: And this one!

File: hippiestation/code/datums/diseases/advance/symptoms/heal.dm [hunk 6]
```diff
@@ -0,0 +1,632 @@
...
+			M.update_damage_overlays()
+
+	return 1
+
+/datum/symptom/heal/water/passive_message_condition(mob/living/M)
+	if(M.getBruteLoss() || M.getFireLoss())
+		return TRUE
+	return FALSE
+
+/datum/symptom/heal/plasma
```
Comment (GuyonBroadway) [inline hippiestation/code/datums/diseases/advance/symptoms/heal.dm]: This one as well!

File: hippiestation/code/datums/diseases/advance/symptoms/heal.dm [hunk 7]
```diff
@@ -0,0 +1,632 @@
...
+		return
+	if(prob(5))
+		to_chat(M, "<span class='notice'>The pain from your wounds fades rapidly.</span>")
+	for(var/obj/item/bodypart/L in parts)
+		if(L.heal_damage(heal_amt/parts.len, heal_amt/parts.len))
+			M.update_damage_overlays()
+	return 1
+
+
+/datum/symptom/heal/radiation
```
Comment (GuyonBroadway) [inline hippiestation/code/datums/diseases/advance/symptoms/heal.dm]: This one as well.

**Review by GuyonBroadway — CHANGES_REQUESTED** [review]

**Comment by GuyonBroadway** [comment]
Instead of copying just the "Healing" and "Better healing" symptoms I'd suggest looking to make them work with thresholds.

**Comment by Jujumatic** [comment]
I'm waiting for Erik to bring the wiki back up so I can get more info on the symptoms, because there's nothing regarding the symptoms, their descriptions and thresholds in oldcode

**Comment by Jujumatic** [comment]
I've also copied some of tg's healing stuff because otherwise what'll end up happening is the symptom pool will have too many symptoms in it, therefore making it horribly difficult to get specific symptoms. So I disabled tg's healing symptom stuff and copied the cool ones over to hippie code

File: hippiestation/code/datums/diseases/advance/symptoms/sensory.dm [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+		M.slurring = max(0, M.slurring - 2)
+		M.confused = max(0, M.confused - 2)
+		if(purge_alcohol)
+			M.reagents.remove_all_type(/datum/reagent/consumable/ethanol, 3)
+			if(ishuman(M))
+				var/mob/living/carbon/human/H = M
+				H.drunkenness = max(H.drunkenness - 5, 0)
+
+	if(A.stage >= 4)
+		M.drowsyness = max(0, M.drowsyness - 2)
```
Comment (JohnGinnane) [inline hippiestation/code/datums/diseases/advance/symptoms/sensory.dm:19]: just to be sure, if you're on stage 3 or above then you will be removing 4 drowsyness, correct?

File: hippiestation/code/datums/diseases/advance/symptoms/heal.dm [hunk 9]
```diff
@@ -0,0 +1,195 @@
...
+	threshold_desc = ""
+
+/datum/symptom/heal/dna/Heal(mob/living/carbon/M, datum/disease/advance/A)
+	var/amt_healed = 1
+	M.adjustBrainLoss(-amt_healed)
+	//Non-power mutations, excluding race, so the virus does not force monkey -> human transformations.
+	var/list/unclean_mutations = (GLOB.not_good_mutations|GLOB.bad_mutations) - GLOB.mutations_list[RACEMUT]
+	M.dna.remove_mutation_group(unclean_mutations)
+	M.radiation = max(M.radiation - (2 * amt_healed), 0)
+	return 1
```
Comment (JohnGinnane) [inline hippiestation/code/datums/diseases/advance/symptoms/heal.dm]: return TRUE

File: hippiestation/code/datums/diseases/advance/symptoms/heal.dm [hunk 10]
```diff
@@ -0,0 +1,195 @@
...
+		M.bodytemperature = min(310, M.bodytemperature + (10 * heal_amt * TEMPERATURE_DAMAGE_COEFFICIENT))
+
+	if(!parts.len)
+		return
+
+	for(var/obj/item/bodypart/L in parts)
+		if(L.heal_damage(0, heal_amt/parts.len))
+			M.update_damage_overlays()
+
+	return 1
```
Comment (JohnGinnane) [inline hippiestation/code/datums/diseases/advance/symptoms/heal.dm]: return TRUE

File: hippiestation/code/datums/diseases/advance/symptoms/heal.dm [hunk 11]
```diff
@@ -0,0 +1,195 @@
...
+	var/list/parts = M.get_damaged_bodyparts(1,1) //1,1 because it needs inputs.
+
+	if(!parts.len)
+		return
+
+	for(var/obj/item/bodypart/L in parts)
+		if(L.heal_damage(0, heal_amt/parts.len))
+			M.update_damage_overlays()
+
+	return 1
```
Comment (JohnGinnane) [inline hippiestation/code/datums/diseases/advance/symptoms/heal.dm]: return TRUE

File: hippiestation/code/datums/diseases/advance/symptoms/heal.dm [hunk 12]
```diff
@@ -0,0 +1,195 @@
...
+		M.take_bodypart_damage(0, 1) //Deals BURN damage, which is not cured by this symptom
+
+	if(!parts.len)
+		return
+
+	for(var/obj/item/bodypart/L in parts)
+		if(L.heal_damage(heal_amt/parts.len, 0))
+			M.update_damage_overlays()
+
+	return 1
```
Comment (JohnGinnane) [inline hippiestation/code/datums/diseases/advance/symptoms/heal.dm]: return TRUE

File: hippiestation/code/datums/diseases/advance/symptoms/heal.dm [hunk 13]
```diff
@@ -0,0 +1,195 @@
...
+	resistance = -2
+	stage_speed = -2
+	transmittable = -2
+	level = 6
+	threshold_desc = ""
+
+/datum/symptom/oldcode/heal/plus/Heal(mob/living/M, datum/disease/advance/A)
+	var/heal_amt = 1
+	M.adjustToxLoss(-heal_amt)
+	return 1
```
Comment (JohnGinnane) [inline hippiestation/code/datums/diseases/advance/symptoms/heal.dm]: return TRUE

File: hippiestation/code/datums/diseases/advance/symptoms/heal.dm [hunk 14]
```diff
@@ -0,0 +1,195 @@
...
+	var/mob/living/M = A.affected_mob
+	switch(A.stage)
+		if(4, 5)
+			Heal(M, A)
+	return
+
+/datum/symptom/oldcode/heal/proc/Heal(mob/living/M, datum/disease/advance/A)
+	var/heal_amt = 0.5
+	M.adjustToxLoss(-heal_amt)
+	return 1
```
Comment (JohnGinnane) [inline hippiestation/code/datums/diseases/advance/symptoms/heal.dm]: return TRUE

File: hippiestation/code/datums/diseases/advance/symptoms/heal.dm [hunk 15]
```diff
@@ -0,0 +1,195 @@
...
+	var/list/parts = M.get_damaged_bodyparts(1,1) //1,1 because it needs inputs.
+
+	if(!parts.len)
+		return
+
+	for(var/obj/item/bodypart/L in parts)
+		if(L.heal_damage(heal_amt/parts.len, 0))
+			M.update_damage_overlays()
+
+	return 1
```
Comment (JohnGinnane) [inline hippiestation/code/datums/diseases/advance/symptoms/heal.dm]: return TRUE

File: hippiestation/code/datums/diseases/advance/symptoms/heal.dm [hunk 16]
```diff
@@ -0,0 +1,195 @@
...
+	threshold_desc = ""
+
+/datum/symptom/oldcode/heal/brute/plus/Heal(mob/living/carbon/M, datum/disease/advance/A)
+	var/heal_amt = 2
+
+	var/list/parts = M.get_damaged_bodyparts(1,1) //1,1 because it needs inputs.
+
+	if(M.getCloneLoss() > 0)
+		M.adjustCloneLoss(-1)
+		M.take_bodypart_damage(0, 1) //Deals BURN damage, which is not cured by this symptom
```
Comment (JohnGinnane) [inline hippiestation/code/datums/diseases/advance/symptoms/heal.dm]: can you use the BURN define because this isn't obvious even with comment

**Review by JohnGinnane — DISMISSED** [review]
🎏

**Comment by JohnGinnane** [comment]
@HippieStation/head need one more +1

File: hippiestation/code/datums/diseases/advance/symptoms/sensory.dm (hunk 8 above)
Comment (Jujumatic) [reply hippiestation/code/datums/diseases/advance/symptoms/sensory.dm:19]: yes this is to remove all drowsyness that the mob has

**Comment by Jujumatic** [comment]
this is fully done now, i will test it later. i just got the inspiration to get descriptions and threshold stuff from tg's old PRs, as our wiki is probably never coming back, and it turns out most of the stuff was there!

**Comment by Jujumatic** [comment]
Tested, the pandemic machine and the reagent bottles seem to cause some runtimes, especially when adding reagents to the bottle that causes the virus to gain a symptom or the like. However these seem like some general virology runtimes rather than being related to this PR, so yeah. The symptoms work as intended based on what I tested, and the TG ones I changed have not popped up so that's good. This is ready for merging

**Comment by Pyko1** [comment]
sure, why not +1

**Comment by earthmoonjune** [comment]
+1

**Comment by Jujumatic** [comment]
Need a maintainer review plssss

**Review by GuyonBroadway — APPROVED** [review]
the code looks ok but these diseases are a little strong. Balance concerns in a later post.

**Comment by Jujumatic** [comment]
ok yeah the runtime is actually an issue with reagents calling add_reagent on null, will fix it at some point but it's an issue with virology in live too

**Pull request merged**